services/src/tasks/in_memory.rs

```diff
@@ -179,8 +319,10 @@ impl TaskContext for SimpleTaskManagerContext {
 
         if task_status.is_running() {
             *task_status = TaskStatus::Running(RunningTaskStatusInfo::new(pct_complete, status));
+        } else if task_status.is_aborting() {
+            *task_status = TaskStatus::Aborting(RunningTaskStatusInfo::new(pct_complete, status));
```

Comment:
why can the  task still make progress if it is aborting?

Reply:
Imagine, you abort the creation of 100 files, you abort and the clean-up is to remove 98 files.
Then, you could increment the progress after each removal.
I thought it would be nice to know when the abortion is over.